src/Symfony/Component/DependencyInjection/Compiler/AutowiringPass.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Guesses constructor arguments of services definitions and try to instantiate services if necessary.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class AutowiringPass implements CompilerPassInterface
+{
+    private $container;
+    private $definitions;
+    private $reflectionClassesToId = array();
+    private $definedTypes = array();
+    private $typesToId;
+    private $notGuessableTypesToId = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $this->container = $container;
+        $this->definitions = $container->getDefinitions();
+        foreach ($this->definitions as $id => $definition) {
+            $this->completeDefinition($id, $definition);
+        }
+    }
+
+    /**
+     * Wires the given definition.
+     *
+     * @param string     $id
+     * @param Definition $definition
+     *
+     * @throws RuntimeException
+     */
+    private function completeDefinition($id, Definition $definition)
+    {
+        if (!($reflectionClass = $this->getReflectionClass($id, $definition))) {
+            return;
+        }
+
+        if (!($constructor = $reflectionClass->getConstructor())) {
+            return;
+        }
+
+        $arguments = $definition->getArguments();
+        foreach ($constructor->getParameters() as $index => $parameter) {
```

Comment:
getParameters returns an array indexed by name IIRC, not by numbers, so the ``$argumentExist`` check will always be false